### PR TITLE
Introduce Block Protocol v0.2 and custom element support

### PR DESCRIPTION
**Pull request by CiaranMn**
This PR introduces the new specification, docs and tooling for version 0.2 of the Block Protocol:
- new spec pages
- new doc pages
- package updates:
  - new `@blockprotocol/graph` package for use by block protocol authors and embedding applications
  - new `@blockprotocol/core` package which is used by the graph package
  - update `mock-block-dock` for the new spec

As part of this it introduces proper custom element / web component support:
- `custom-element` as a specified `blockType`
- new `custom-element` template via `create-block-app` 
- `mock-block-dock` support for custom elements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +61,7 @@ If you’re looking for inspiration regarding new blocks to build, or contributi
 
 Distributed under the MIT License. See `LICENSE.md` for more information.
 
-## Contact
+........................................................................................................................## Contact
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply README.md]: :D fixed

File: packages/block-template/templates/custom-element/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+  "name": "custom-element-block",
+  "version": "0.0.0",
+  "description": "Block Protocol block template for a custom element-based block",
```
Comment (Alfred-Mountfield) [inline packages/block-template/templates/custom-element/package.json:4]:
```suggestion
  "description": "Block Protocol block template for a custom element-based block",
```
```suggestion
  "description": "Block Protocol block template for a custom-element based block",
```

Comment (CiaranMn) [reply packages/block-template/templates/custom-element/package.json:4]: I guess we could go for a triple hyphen?

Comment (Alfred-Mountfield) [reply packages/block-template/templates/custom-element/package.json:4]: Sorry gotcha, I was going off these things 
image
Didn't realise they were also slugs.

I'm good with no hyphens, or triple hyphen. But it's not too important

**Review by nonparibus — DISMISSED** [review]

File: packages/block-template/templates/custom-element/package.json (hunk 2 above)
Comment (nonparibus) [reply packages/block-template/templates/custom-element/package.json:4]: Excuse me, but _I am_ the worst at hyphens, as Ciaran's suggestions on my prior PRs this last fortnight clearly demonstrate

**Review by nonparibus — APPROVED** [review]

**Review by nonparibus — APPROVED** [review]

**Pull request merged**